addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/internal/JeeLinkHandler.java

```diff
@@ -0,0 +1,161 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.jeelink.internal;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.BridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.jeelink.internal.config.JeeLinkConfig;
+import org.openhab.binding.jeelink.internal.connection.AbstractJeeLinkConnection;
+import org.openhab.binding.jeelink.internal.connection.ConnectionListener;
+import org.openhab.binding.jeelink.internal.connection.JeeLinkConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for a JeeLink USB Receiver thing.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class JeeLinkHandler extends BaseBridgeHandler implements BridgeHandler, ConnectionListener {
+    private final Logger logger = LoggerFactory.getLogger(JeeLinkHandler.class);
+
+    private JeeLinkConnection connection;
+    private Map<String, JeeLinkReadingConverter> converters = new HashMap<>();
+    private Map<JeeLinkReadingConverter, List<ReadingHandler>> convSensorMap = new HashMap<>();
+
+    private AtomicBoolean connectionInitialized = new AtomicBoolean(false);
+    private ScheduledFuture<?> connectJob;
+
+    public JeeLinkHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        JeeLinkConfig cfg = getConfig().as(JeeLinkConfig.class);
+
+        try {
+            for (JeeLinkReadingConverter c : SensorDefinition.createConverters(this)) {
+                converters.put(c.getSketchName(), c);
+            }
+
+            connection = AbstractJeeLinkConnection.createFor(cfg, scheduler, this);
+            connection.openConnection();
+        } catch (java.net.ConnectException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void connectionOpened() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void connectionAborted(String cause) {
+        Runnable connector = new Runnable() {
+            @Override
+            public void run() {
+                connection.openConnection();
+            }
+        };
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, cause);
+
+        connectJob = scheduler.schedule(connector, 10, TimeUnit.SECONDS);
```

Comment:
That's not fair. Java 8 was not even released when I created the pull request! ;-)